hashicorp/dev-portal
# feat: opt out form submits

## ✅ "Ready for Review" checklist




---

## 🔗 Relevant links



- [Preview link](https://dev-portal-git-ksopt-out-form-submits-hashicorp.vercel.app/) 🔎
- [Asana task](https://app.asana.com/0/1199634971449915/1202349556269721) 🎟️

## 🗒️ What

This PR handles the form submission leveraging the api route introduced via #486. 

## 🧪 Testing

- Visit the preview [with the opt param](https://dev-portal-git-ksopt-out-form-submits-hashicorp.vercel.app/waypoint/tutorials?optInFrom=learn) to set the cookie. 
- Click the leave beta button
- Fill out the form
- [Check the google sheet ](https://docs.google.com/spreadsheets/d/15KyBwiEui6YfatLkN0QXcee3TR-pew_sVhnb9nk0DWo/edit#gid=0)to verify the data was submitted

## Additional

It'd eventually be nice to add a 'pending' state while we're waiting for that form submission to come through before the redirect occurs. I'll leave that as a [backlog thing for now. ](https://app.asana.com/0/1202097197789424/1202348106428666)

_Note_ if you haven't opted in on the learn platform and you manually hit the 'optInFrom' route, the redirect on the opt out won't work properly.
